### ISSUE TITLE
TimePicker with short time pattern shows verbose time

### ISSUE THREAD

**Issue by breyed** [xamarin/Xamarin.Forms#3306]
On WPF, if you set `TimePicker.Format` to "t" ([short time](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#ShortTime)), and click in the TimePicker, text like `08:37:20.0544050` appears. Instead, the TimePicker should display a time like the output of `DateTime.Now.ToString("t")`, e.g. `8:47 AM`.

Xamarin.Forms 3.1.0.637273

**Comment by mohachouch** [comment]
Hi @breyed,

The timepicker use a TimeSpan or the "t" format is only for an DateTime.

You can use all custom timespan format : https://docs.microsoft.com/fr-fr/dotnet/standard/base-types/custom-timespan-format-strings

@rmarinho @PureWeen I don 't know how the timeformat is processed on other platforms. Does it allow the patterns of datetime ?

**Comment by breyed** [comment]
@mohachouch Your English was hard to follow, but it looks like you're suggesting that `TimePicker` uses the format strings for `TimeSpan`. However, the docs for [TimePicker.Format](https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.timepicker.format?view=xamarin-forms#Xamarin_Forms_TimePicker_Format) say that it uses the format strings for `DateTime`. That makes sense to me, since a time picker represents a time, which has elements like AM and PM, which a time span does not.

**Comment by mohachouch** [comment]
@breyed Thanks PR soon
